core/lib/dal/src/tee_verifier_input_producer_dal.rs

```diff
@@ -72,9 +72,63 @@ impl TeeVerifierInputProducerDal<'_, '_> {
         Ok(())
     }
 
+    pub async fn get_new_l1_batch(&mut self) -> DalResult<Option<L1BatchNumber>> {
+        // Since we depend on Merkle paths, we use the proof_generation_details table to inform us of newly available to-be-proven batches.
+        let row = sqlx::query!(
+            r#"
+            SELECT
+                MAX(l1_batch_number) AS "number"
+            FROM
+                proof_generation_details
```

Comment:
What code do you intend to duplicate as a potential solution? I'm open to suggestions in terms of an alternate "signal" to monitor for new batches.

Reply:
I think that the right way would be to move `save_merkle_paths_artifacts_metadata` from `proof_generation_dal` to the `blocks_dal` (e.g. existence of the metainformation is the property of the block, regardless of the consumer).

This way we (later, not now!) will be able to get rid of `insert_proof_generation_details` in the `updater.rs` as well -- so that the tree is not aware of any verification logic (as it should be).